src/riscv/lib/src/pvm/linux.rs

```diff
@@ -1009,6 +1013,21 @@ impl<M: ManagerBase> SupervisorState<M> {
         // Return 0 as an indicator of success
         Ok(0)
     }
+
+    /// Handle `prlimit64` system call.
+    ///
+    /// See: <https://www.man7.org/linux/man-pages/man2/prlimit64.2.html>
+    fn handle_prlimit64(&self) -> Result<u64, Error> {
+        // In jstz this is only used to set the maximum size of the process stack to infinity. In
+        // other programs, it can be used to set limits for system resources. Allowing programs to
+        // set these values to extreme values would inflate proof size, so is undesirable.
+        // Futhermore, a privileged process (such as the supervisor implemented in this PVM) has
+        // the ability to set these limits to arbitrary values.
+        // So it does! By ignoring this system call, and using its own predefined limits.
```

Comment:
What about other uses of `prlimit64`? What if this system call is used for reading?

It is not enough to assume that all kernels will only ever do what one version of Jstz does. So either you filter and allow only one use, or you try to support all uses. Since there are only 16 resource types, the latter should be viable as well.